cryptography/src/ed25519.rs

```diff
@@ -108,11 +111,290 @@ impl Scheme for Ed25519 {
             Err(_) => return false,
         };
         let signature = ed25519_consensus::Signature::from(signature);
-        let payload = union_unique(namespace, message);
-        public_key.verify(&signature, &payload).is_ok()
+        match namespace {
+            Some(namespace) => {
+                let payload = union_unique(namespace, message);
+                public_key.verify(&signature, &payload).is_ok()
+            }
```

Comment:
nit: maybe one-lining it could make it easier to compare to the `None` case? This is what you've done above on line 81. But it's a style preference
```suggestion
            Some(namespace) => public_key.verify(&signature, &union_unique(namespace, message)).is_ok(),
```